### DISCUSSION TITLE
Priority updates for Skeleton Labs

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#2729] [Announcements]
Hey everyone, Chris here from [Skeleton Labs](https://www.skeletonlabs.co/).

As you may know, Skeleton went open source in early August 2022. Over the last two years, we've seen a huge influx of new contributors and community members, launched multiple major releases to the core project, began working on the [next generation of Skeleton](https://next.skeleton.dev/), as well as our recent announcement of [Floating UI Svelte](https://floating-ui-svelte.vercel.app/).

I wanted to take a moment to give everyone in the community a quick status update regarding all Skeleton Labs projects. Starting this week, you may observe some changes that will impact the priorities and release schedule for several of our projects.

## Why are Changes Needed?

To provide some context, it's important to understand that while Skeleton is a moderately sized open source project, and we have a number of contributors, the overwhelming majority of planning, updates, PR reviews, and triage come directly from, well, [myself](https://github.com/skeletonlabs/skeleton/graphs/contributors). Over the last few months I've shifted from working on Skeleton full-time, to now operating part-time alongside a day job. This means I have roughly around 20 hours a week to contribute directly to Skeleton. Though I often provide support well beyond that.

Currently I contribute to four separate projects: my day job, [Skeleton v2](https://www.skeleton.dev/), [Skeleton v3](https://next.skeleton.dev/), and [Floating UI Svelte](https://floating-ui-svelte.vercel.app/). While I love working on all these projects, lately my work life balance and ability to actually remain productive has taken a hit. To put it plainly, I've been stretched a bit thin, so we're going to make a few adjustments.

## Feature Freeze for Skeleton v2

First up, the team and I are feeling great about the maturity of Skeleton v2. While it's not perfect, we have already started tackling the largest known issues as part of the Skeleton v3 updates. As such, we're implementing a [feature freeze](<https://en.wikipedia.org/wiki/Freeze_(software_engineering)>) for v2. This will come with two notable exceptions:

1. Breaking issues
2. Svelte v5 compatibility issues

Beyond this, we still welcome and encourage community-provided contributions for bugfixes and minor feature improvements. We do ask that major updates and changes please be [diverted toward Skeleton v3](https://github.com/skeletonlabs/skeleton/milestone/5).

## Prioritizing Skeleton v3

Going forward, myself and the team will prioritize Skeleton v3, starting with components. Today's new release actually introduces [three new v3 components](https://github.com/skeletonlabs/skeleton/discussions/2728). And we hope to continue or increase this pattern going forward. Once most components are fulfilled, we'll turn our attention to integration guides, refinement, and polish. Anything that helps get us close the gap towards release.

## Changes to Technical Support

Were lucky to have a number of wonderful moderators and contributors that actively assist with technical support, both on GitHub and Discord. However, as the community continues to grow, so does the number of technical support issues. To set realistic expectations going forward, the team (and myself especially) may take a step back from this. There may be a slower response time. Transitioning from essentially 24/7 support, to more of voluntary basis, every few days, etc. We feel this will provide a healthier balance of our time, so we appreciate your patience going forward! If you need help though, please do reach out. That's why we provide support!

## Updates for Floating UI

No major changes here. @Hugos68 will continue to operate as the core maintainer for the project, while I contribute on a voluntary basis, and as time allows. Please note that this project is a pre-requisite for v3 to launch, so do expect priority to return to this in the near future.

## Fewer Video Updates

While we've been thrilled to see a huge uptick in our [YouTube community](https://www.youtube.com/@skeletonlabs), we will be pausing video updates for a little while. While helpful for visual folks (like myself!) these take a lot of time to produce. Even a 15-20 minute video can take upwards of a whole afternoon between prep, execution, rendering, and upload. Going forward, videos will be less frequent, but likely cover a larger spectrum of updates. In the meantime, we'll continue to announce notable additions on our social channels, leaning more on the v3 doc site to explain each feature in detail.

## Taking a Short Break

If this post has not made it clear, I am extremely bad at prioritizing time off for myself. So I'll be taking a week long break from June 21st through July 1st. During this time I will be checking in and contributing in a very minimal fashion. My plan is to enjoy some R&R (play a bit of the [Elden Ring DLC](https://www.youtube.com/watch?v=J9Bm5U-MJZw)) then return refreshed before we kick off this final phase leading to the v3 launch. Again I appreciate everyone's patience and understanding, and ask that you please reach out to other members of the team if you need immediate support during this time.

## Our Goal

Basically we're trying to streamline our limited time and resources to ensure we're putting the most focus on Skeleton v3 going forward. With the hope of launching alongside the first Svelte v5 stable release. In the meantime, if you would like to continue to see updates to other projects, please consider contributing directly! We always welcome the additional help and support! After all, this is what open source is all about.

Beyond that, I want to once again thank everyone for continuing to make Skeleton such a rewarding project to work on. I hope myself and the team can continue to have the privilege of building useful open source software for you for many years to come!

Cheers,
Chris
